### ISSUE TITLE
Couldn’t communicate with a helper application. Disk space is very low

### ISSUE THREAD

**Issue by loonix** [miguelpruivo/flutter_file_picker#885]
**Describe the bug**
I have encountered a difficulty trying to use the file picker when the phone has no storage (which will be fine for most people but my ipad is full of junk).

**Platform**
- [] Android - have not tested in Android but believe will be similar
- [X] iOS
- [] Web
- [] Desktop (Go)

**Platform OS version**
What version did it happen?
14

**How are you picking?**
```
 var image = await FilePicker.platform.pickFiles(allowMultiple: false, type: FileType.image);
  widget.imagePicked!(image!.files[0]);
```

**Details to reproduce the issue**
You need to have low storage on your IOS device, then select a photo that is on the cloud and add it with the file picker

**Error Log**
```
***** Picker:<PHPickerViewController: 0x1036ff3d0> didFinishPicking:(
    "<PHPickerResult: 0x282649cc0>"
)
* -[FilePickerPlugin picker:didFinishPicking:] [Line 389]
Error copying file type public.item. Error: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.item" UserInfo={NSLocalizedDescription=Cannot load representation of type public.item, NSUnderlyingError=0x2828f2f40 {Error Domain=NSCocoaErrorDomain Code=4101 "Couldn’t communicate with a helper application." UserInfo={NSUnderlyingError=0x2828f3bd0 {Error Domain=PHAssetExportRequestErrorDomain Code=3 "The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.)" UserInfo={NSLocalizedDescription=The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.), NSUnderlyingError=0x2828f2d60 {Error Domain=CloudPhotoLibraryErrorDomain Code=1005 "Disk space is very low" UserInfo=0x28267e120 (not displayed)}}}}}}
***** Could not load the picked given file: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.item" UserInfo={NSLocalizedDescription=Cannot load representation of type public.item, NSUnderlyingError=0x2828f2f40 {Error Domain=NSCocoaErrorDomain Code=4101 "Couldn’t communicate with a helper application." UserInfo={NSUnderlyingError=0x2828f3bd0 {Error Domain=PHAssetExportRequestErrorDomain Code=3 "The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.)" UserInfo={NSLocalizedDescription=The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.), NSUnderlyingError=0x2828f2d60 {Error Domain=CloudPhotoLibraryErrorDomain Code=1005 "Disk space is very low" UserInfo=0x28267e120 (not displayed)}}}}}}
* -[FilePickerPlugin picker:didFinishPicking:]_block_invoke [Line 413]
flutter: RangeError (index): Invalid value: Valid value range is empty: 0
```

**Flutter Version details**
v2.2.1

beware I am using version:
  file_picker: ^4.1.3

**Comment by loonix** [comment]
Updated to latest (4.2.3)

Updated terminal error:
```
***** Picker:<PHPickerViewController: 0x125f9db50> didFinishPicking:(
    "<PHPickerResult: 0x280b4ffc0>"
)
* -[FilePickerPlugin picker:didFinishPicking:] [Line 401]
Error copying file type public.item. Error: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.item" UserInfo={NSLocalizedDescription=Cannot load representation of type public.item, NSUnderlyingError=0x280500180 {Error Domain=NSCocoaErrorDomain Code=4101 "Couldn’t communicate with a helper application." UserInfo={NSUnderlyingError=0x2805034e0 {Error Domain=PHAssetExportRequestErrorDomain Code=3 "The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.)" UserInfo={NSLocalizedDescription=The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.), NSUnderlyingError=0x2805012f0 {Error Domain=CloudPhotoLibraryErrorDomain Code=1005 "Disk space is very low" UserInfo=0x280b094e0 (not displayed)}}}}}}
***** Could not load the picked given file: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.item" UserInfo={NSLocalizedDescription=Cannot load representation of type public.item, NSUnderlyingError=0x280500180 {Error Domain=NSCocoaErrorDomain Code=4101 "Couldn’t communicate with a helper application." UserInfo={NSUnderlyingError=0x2805034e0 {Error Domain=PHAssetExportRequestErrorDomain Code=3 "The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.)" UserInfo={NSLocalizedDescription=The operation couldn’t be completed. (PHAssetExportRequestErrorDomain error 3.), NSUnderlyingError=0x2805012f0 {Error Domain=CloudPhotoLibraryErrorDomain Code=1005 "Disk space is very low" UserInfo=0x280b094e0 (not displayed)}}}}}}
* -[FilePickerPlugin picker:didFinishPicking:]_block_invoke [Line 425]
flutter: RangeError (index): Invalid value: Valid value range is empty: 0
```

**Comment by miguelpruivo** [comment]
Fixed with 4.2.4.

**Comment by sproutlyCaleb** [comment]
I got this error as well.  I resolved it by clearing spaced on my device.


* -[FilePickerPlugin picker:didFinishPicking:] [Line 401]
Error copying file type public.item. Error: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.item" UserInfo={NSLocalizedDescription=Cannot load representation of type public.item, NSUnderlyingError=0x282b855c0 {Error Domain=NSCocoaErrorDomain Code=4101 "Couldn’t communicate with a helper application." UserInfo={NSUnderlyingError=0x282b84ea0 {Error Domain=PHAssetExportRequestErrorDomain Code=3 "(null)" UserInfo={NSUnderlyingError=0x282b85800 {Error Domain=CloudPhotoLibraryErrorDomain Code=1005 "Disk space is very low" UserInfo=0x2824f3660 (not displayed)}}}}}}

***** Could not load the picked given file: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.item" UserInfo={NSLocalizedDescription=Cannot load representation of type public.item, NSUnderlyingError=0x282b855c0 {Error Domain=NSCocoaErrorDomain Code=4101 "Couldn’t communicate with a helper application." UserInfo={NSUnderlyingError=0x282b84ea0 {Error Domain=PHAssetExportRequestErrorDomain Code=3 "(null)" UserInfo={NSUnderlyingError=0x282b85800 {Error Domain=CloudPhotoLibraryErrorDomain Code=1005 "Disk space is very low" UserInfo=0x2824f3660 (not displayed)}}}}}}
* -[FilePickerPlugin picker:didFinishPicking:]_block_invoke [Line 428]
Error copying file type public.item. Error: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.item" UserInfo={NSLocalizedDescription=Cannot load representation of type public.item, NSUnderlyingError=0x282b7fa20 {Error Domain=NSCocoaErrorDomain Code=4101 "Couldn’t communicate with a helper application." UserInfo={NSUnderlyingError=0x282b87150 {Error Domain=PHAssetExportRequestErrorDomain Code=3 "(null)" UserInfo={NSUnderlyingError=0x282b84120 {Error Domain=CloudPhotoLibraryErrorDomain Code=1005 "Disk space is very low" UserInfo=0x2824f3b00 (not displayed)}}}}}}
